### ISSUE TITLE
XXX failure of TensorRT X.Y when running XXX on GPU XXX

### ISSUE THREAD

**Issue by loryruta** [NVIDIA/TensorRT#4362]
## Description

I'm having a similar problem in C++ using TensorRT and CUDA. My onnx model has 2 inputs and one output. My inference code is:

```cpp
m_context = std::unique_ptr<nvinfer1::IExecutionContext>(m_engine->createExecutionContext());

// Set input dimensions (as input shape is dynamic)
m_context->setInputShape("im0", dims);
m_context->setInputShape("im1", dims);

status = m_context->setTensorAddress("im0", im0_device_ptr), assert(status);
status = m_context->setTensorAddress("im1", im1_device_ptr), assert(status);
status = m_context->setTensorAddress("disparity_map", out_disparity_map_device_ptr), assert(status);

status = m_context->enqueueV3(stream), assert(status);
```

The error I'm getting is:

```
[ERROR] [] TensorRT error: IExecutionContext::enqueueV3: Error Code 1: Cask (Cask Pooling Runner Execute Failure)
```

I've also tried changing enqueueV3 to executeV2 - still same issue.

I'm 99.99% sure `im0_devce_ptr`, `im1_device_ptr` and `out_disparity_map_device_ptr` are valid pointers to device memory.

The thing is, I have the exact similar code in Python + PyTorch and it works using **the same engine**.

## Environment

**TensorRT Version**: 10.8

**NVIDIA GPU**: RTX 2060 SUPER

**NVIDIA Driver Version**: 560.35.05

**CUDA Version**: 12.6

**CUDNN Version**: -

Operating System: Ubuntu 24.04

Python Version (if applicable): 3.11

Tensorflow Version (if applicable):

PyTorch Version (if applicable): 2.5.1

Baremetal or Container (if so, version): Baremetal

## Relevant Files

**Model link**:

https://drive.google.com/file/d/108rgI-m2-3Xg17vRDdvXf3yS-dX8fFHV/view?usp=drive_link

## Steps To Reproduce

**Commands or scripts**:

**Have you tried [the latest release](https://developer.nvidia.com/tensorrt)?**:

**Can this model run on other frameworks?** For example run ONNX model with ONNXRuntime (`polygraphy run <model.onnx> --onnxrt`):

**Comment by kevinch-nv** [comment]
I see the model contains dynamic shapes, what are the optimization profile shape ranges, and the shape of the actual inference input that's causing the exception?

**Comment by loryruta** [comment]
> I see the model contains dynamic shapes, what are the optimization profile shape ranges, and the shape of the actual inference input that's causing the exception?

Hi! Thanks for your answer 😄 

I actually came up myself with the issue ([I have also opened a parallel thread on the NVIDIA forum](https://forums.developer.nvidia.com/t/help-with-tensorrt-errors-when-building-an-engine/324552)).

It was caused by the fact that internally the model expects its dynamic input shapes (2nd and 3rd dim; image H and W) to be divisible by 32. After shaping the input correctly TensorRT worked like charm!
